internal/worker/json.go

```diff
@@ -76,6 +76,39 @@ func (j *OSBuildJobResult) TargetErrors() []*clienterrors.Error {
 	return targetErrors
 }
 
+// RpmErrors a *clienterrors.Error gathered
+// from the failed rpm stage within the job's error.
+// If there was no RPM error then nil is returned.
+func (j *OSBuildJobResult) RpmError() *clienterrors.Error {
```

Comment:
Let's just return a string and embed that in the details. If we just embed a string in the details image builder will automatically forward it. I'm not sure nested error structs work without explicit changes in https://github.com/osbuild/image-builder/blob/main/internal/v1/handler.go#L262

Reply:
I'm not sure I understand either, what do you mean by nested ? The function `RpmError` returns a single error with for details the output which is a text string. The same way we generate an error on line 305.